rplidar.go

```diff
@@ -35,16 +32,47 @@ const (
 // Model is the model of the rplidar
 var Model = resource.NewModel("viam", "lidar", "rplidar")
 
+// Rplidar controls an Rplidar device.
+type Rplidar struct {
+	resource.Named
+	resource.AlwaysRebuild
+	mu                      sync.Mutex
+	device                  rplidarDevice
+	nodes                   gen.Rplidar_response_measurement_node_hq_t
+	nodeSize                int
+	started                 bool
+	scannedOnce             bool
+	defaultNumScans         int
+	warmupNumDiscardedScans int
+
+	logger golog.Logger
+}
+
+// Config describes how to configure the RPlidar component.
+type Config struct {
+	DevicePath string `json:"device_path"`
+}
+
+// Validate checks that the config attributes are valid for an RPlidar.
+func (conf *Config) Validate(path string) ([]string, error) {
+	return nil, nil
+}
+
 func init() {
-	registry.RegisterComponent(camera.Subtype, Model, registry.Component{Constructor: newRplidar})
+	resource.RegisterComponent(camera.API, Model, resource.Registration[camera.Camera, *Config]{Constructor: newRplidar})
```

Comment:
we weren't using an AttributeMapConverter before, but went ahead and added one here because this interface makes it easier to provide an AttributeMapConverter than not